src/applications/search/containers/SearchApp.jsx

```diff
@@ -94,6 +94,17 @@ class SearchApp extends React.Component {
     if (e) e.preventDefault();
     const { userInput, currentResultsQuery, page } = this.state;
 
+    let userInputFromAddress = '';
+    let pageFromAddress;
+
+    if (this.props.router.location.query) {
+      userInputFromAddress = this.props.router.location.query.query;
+      pageFromAddress = this.props.router.location.query.page;
```

Comment:
Actually, I think you can write this like:

```jsx
const userInputFromAddress = this.props.router?.location?.query?.query;
const pageFromAddress = this.props.router?.location?.query?.page;
```
without the `if` statement if you want, since it will be null if the query doesn't exist

Reply:
Would saying `[userInput/page]FromURL` instead of `FromAddress` be clearer? We don't really mess with address data here but just trying to avoid a future dev thinking that it means a physical address, since that's what my brain would do for a good minute lol

Reply:
These are some great comments, frankly I copied some existing code from higher up in the component, so I would be happy to make some edits!